File: 2022/Day1/day1.py
# %%
import os
import numpy as np
os.chdir('/Users/tobiasmolenaar/Documents/GitHub/AdventofCode/2022/Day1/')

f = open('input_day1.txt', 'r')
data = f.read().split('\n')
f.close()

elvesCalories = np.zeros(len(data))
elfNumber = 0
for calories in data:
    if calories == '':
        elfNumber += 1
    else:
        elvesCalories[elfNumber] += int(calories)

        
# Part 1
print(max(elvesCalories))

# Part 2
print(sum(sorted(elvesCalories)[-3:]))

File: 2024/Day01/day1.jl
file = "/Users/tobiasmolenaar/Documents/GitHub/AdventofCode/2024/Day01/input_day1.txt"

using DelimitedFiles
x = readdlm(file, Int)

# part 1
print(sum(abs.(sort(x[:, 2]) - sort(x[:, 1]))))

# part 2
similarity = 0
for value in x[:, 1]
    occurrence = count(isequal(value), x[:, 2])
    global similarity += occurrence * value
end
print("\n")
print(similarity)